alloc.h

```diff
@@ -31,23 +31,10 @@
 #ifndef HIREDIS_ALLOC_H
 #define HIREDIS_ALLOC_H
 
-#include <stdlib.h> /* for size_t */
-
-#ifndef HIREDIS_OOM_HANDLER
-#define HIREDIS_OOM_HANDLER abort()
-#endif
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-void *hi_malloc(size_t size);
-void *hi_calloc(size_t nmemb, size_t size);
-void *hi_realloc(void *ptr, size_t size);
-char *hi_strdup(const char *str);
-
-#ifdef __cplusplus
-}
-#endif
+#define hi_malloc malloc
+#define hi_calloc calloc
+#define hi_realloc realloc
+#define hi_strdup strdup
+#define hi_free free
```

Comment:
Sure, that works for me.  It would make it much simpler for end-users to plug in their own functions too.

Reply:
Updated the logic to use a single pass-through struct with function pointers and just access the appropriate member depending on what kind of allocation it is.

I went with static inline functions here because it allowed me to have the allocation functions configurable without paying a performance price.  Using non-inline functions defined in `alloc.c` cost between 8-15% on my machine for some workloads.